app/controllers/tags_controller.rb

```diff
@@ -1,22 +1,16 @@
 class TagsController < ApplicationController
-
   def index
-    @tags =  Tag.search(params[:q])
-  end
-
-  def popular
-    render json: Tag.get_tags_that_are(:popular), status: 200
-  end
-
-  def recent
-    render json: Tag.get_tags_that_are(:recent), status: 200
+    render partial: 'tag_properties', locals: { tags: Tag.search(params[:q])}
```

Comment:
@andela-echigbo a cursory look at the two partial will reveal that they serialise their payload differently and both partials _by extension the serialisation they offer_ are currently actively required _(at the least pending the [resolution of this suggestion](https://andela.slack.com/archives/zhishi-engine/p1461227332000005) by @andela-oakinniranye)_